pallets/stellar-relay/src/lib.rs

```diff
@@ -601,15 +601,21 @@ pub mod pallet {
 			// We store the externalized value in a variable so that we can check if it's the same
 			// for all envelopes. We don't distinguish between externalized and confirmed values as
 			// it should be the same value regardless.
-			let mut externalized_value: Option<Value> = None;
-			let mut externalized_n_h: Option<u32> = None;
+			let (externalized_value, externalized_n_h) =
+				match &externalized_envelope.statement.pledges {
+					ScpStatementPledges::ScpStExternalize(externalized_statement) =>
+						(&externalized_statement.commit.value, externalized_statement.n_h),
+					ScpStatementPledges::ScpStConfirm(confirmed_statement) =>
```

Comment:
This case cannot happen as `externalized_envelope` is define in line 567 to be an `ScpStExternalize` message.